apps/www/components/Frame/Header.js

```diff
@@ -190,9 +190,24 @@ const Header = ({
                 )}
                 inNativeIOSApp={inNativeIOSApp}
                 onClick={() => {
-                  if (router.asPath === userButtonLink) {
-                    closeHandler()
+                  if (router.asPath.startsWith(userButtonLink)) {
+                    return closeHandler()
                   } else {
+                    if (userButtonLink === '/anmelden') {
+                      try {
+                        const url = new URL(
+                          userButtonLink,
+                          window.location.origin,
+                        )
+                        url.searchParams.set(
+                          'redirect',
+                          encodeURIComponent(router.asPath),
+                        )
+                        return router.push(url)
+                      } catch (e) {
+                        console.error(e)
```

Comment:
Do we need to log this error? Shouldn't we just `router.push(userButtonLink)` here?